Haskell/syntax_test_haskell.hs

```diff
@@ -145,6 +163,62 @@
    countTheBeforeVowel = undefined
 
 
+--KEYWORDS
+
+import qualified Data.Vector.Mutable as MutableVector
+-- ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ meta.import.haskell
+-- ^^^ keyword.control.import.haskell
+--     ^^^^^^^^^ keyword.control.import.haskell
+--               ^^^^^^^^^^^^^^^^^^^ support.other.module.haskell
+--                                   ^^ keyword.control.import.haskell
+--                                      ^^^^^^^^^^^^^ support.other.module.haskell
+import Data.List.Split (splitOn)
+-- ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ meta.import.haskell
+-- ^^^ keyword.control.import.haskell
+--     ^^^^^^^^^^^^^^^ support.other.module.haskell
+--                     ^^^^^^^^^ meta.declaration.exports.haskell
+--                     ^ punctuation.section.group.begin.haskell
+--                      ^^^^^^^ variable.function.haskell
+--                             ^ punctuation.section.group.end.haskell
+
+test =
+--   ^ keyword.operator.haskell
+    let x = 2 in x * y
+--  ^^^ keyword.declaration.variable.haskell
+--            ^^ keyword.control.context.haskell
+    where
+--  ^^^^^ keyword.control.context.haskell
+        y = 1
+--        ^ keyword.operator.haskell
+
+test a = case a of
+--       ^^^^ keyword.control.conditional.select.haskell
+--              ^^ keyword.control.conditional.select.haskell
+    Nothing -> 0
```

Comment:
Are the arrows scoped? They should be.

Reply:
They are matched by the operator pattern.

Reply:
I think it's probably worth scoping them specifically, since operators should not be scoped as `keyword`, but `->` *should*.

Reply:
I agree, `->`should be a proper keyword, much like guards are. It is used in many different situations and you can't assign to it.